larpandoracontent/LArThreeDReco/LArCosmicRay/CosmicRayTrackRecoveryAlgorithm.cc

```diff
@@ -196,6 +196,11 @@ void CosmicRayTrackRecoveryAlgorithm::MatchClusters(const Cluster* const pSeedCl
     {
         const Cluster *const pTargetCluster = *tIter;
 
+        const std::set daughterVolumeIntersection(LArGeometryHelper::GetCommonDaughterVolumes(pSeedCluster, pTargetCluster));
+        if (daughterVolumeIntersection.empty())
+            continue;
```

Comment:
We could consider throwing an exception if an intersection is not found here.

Reply:
I think in this case `continue` appears to be the right approach, as the code appears to be deciding whether or not to consider the seed and target clusters for matching. An exception here would terminate the function for all subsequent targets, which presumably is not the desired behaviour?

Reply:
I was looking at what happens in the rest of the code - where exceptions are thrown just a few lines below. 

(Regardless of whether we leave this check as it is, I would perhaps move it down after the check at lines 204/205!)